compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -468,6 +502,26 @@ void ConfigurationAdminImpl::NotifyConfigurationUpdated(const std::string& pid)
         }
       }
     }
+    if (pid.find('~') != std::string::npos) {
+      //this is a factory pid
+      fPid = pid;
+    } else {
+      nonFPid = pid;
+    }
+    const auto configurationListeners = configListenerTracker.GetServices();
+    auto type =
+      removed
+        ? cppmicroservices::service::cm::ConfigurationEventType::CM_DELETED
+        : cppmicroservices::service::cm::ConfigurationEventType::CM_UPDATED;
+
+    auto configAdminRef = cmContext.GetServiceReference<ConfigurationAdmin>();
+    for (auto it : configurationListeners) {
+      auto configEvent =
+        std::make_unique<cppmicroservices::service::cm::ConfigurationEvent>(
```

Comment:
Jeff: no need to call "make_unique", since the function doesn't take in a unique pointer